### DISCUSSION TITLE
Performance compared to Supabase

### DISCUSSION THREAD

**Discussion by DePasqualeOrg** [neondatabase/neon#2423] [Q&A] [answered]
I've been testing out Neon on a new project, and after switching to Supabase, I noticed that read queries in particular execute much faster compared to Neon. Is this to be expected, or something that will improve as the project develops?

**Comment by hlinnaka** [comment, marked as answer]
We'd prefer Neon to be faster, of course :-). Many things could affect the latency:

What region is the client on? Neon is currently deployed on AWS us-west-2 region. If you're executing queries from Europe, for example, you will see an additional 150 ms latency on every query, just because of the round trip across the ocean. We're working on deploying on more regions, Europe will be among the first.

What exactly is the query and the schema? Do you have proper indexes in place? Neon runs PostgreSQL, so all the usual PostgreSQL optimization techniques apply. See https://wiki.postgresql.org/wiki/Slow_Query_Questions. One known issue with Neon is that Sequential Scans are slow on Neon, unless the table fits in cache. We're working on that, but indexes could help with that too.

If it's neither of those things, I'd love to hear more about the queries and how you're testing it. It's possible there is some Neon-specific performance issue we've missed...

**Reply by DePasqualeOrg** [reply]
Thanks for your response. I took the AWS region into account and used the same region on Supabase for comparison. I'll see if I can put a minimal demo together to demonstrate what I was observing.

**Comment by jamezening** [comment]
Hi,

Just set up [Neon with Vercel serverless edge](https://neon.tech/docs/serverless/serverless-driver#neon-serverless-driver-with-vercel-edge-functions) - both set in the Singapore region.
I am making requests from South Korea and seeing 5-10 second cold starts and anywhere from 300-500ms latency on non-cold start responses.

The response is just the timestamp as a payload shown in the link.

Is this the expected current performance?
Do you have plans to expand into South Korea, east asian regions?

Thank you.
